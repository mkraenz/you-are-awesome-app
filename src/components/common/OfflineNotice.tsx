import React, { FC } from "react";
import { useTranslation } from "react-i18next";
import { StyleSheet, View } from "react-native";
import { Text, useTheme } from "react-native-paper";

const styles = StyleSheet.create({
    container: {
        marginHorizontal: -100, // ensure full width
        height: 30,
        justifyContent: "center",
        alignItems: "center",
        flexDirection: "row",
    },
    text: {
        color: "white",
    },
});

const OfflineNotice: FC<{}> = () => {
    const { t } = useTranslation();
    const theme = useTheme();
    const containerStyle = {
        ...styles.container,
        backgroundColor: theme.colors.error,
    };
    return (
        <View style={containerStyle}>
            <Text style={styles.text}>{t("noInternet")}</Text>
        </View>
    );
};

export default OfflineNotice;
