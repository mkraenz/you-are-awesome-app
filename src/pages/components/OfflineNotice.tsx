import React, { FC } from "react";
import { StyleSheet, Text, View } from "react-native";

const OfflineNotice: FC<{}> = () => (
    <View style={styles.offlineContainer}>
        <Text style={styles.offlineText}>No Internet Connection</Text>
    </View>
);

const styles = StyleSheet.create({
    offlineContainer: {
        backgroundColor: "#b52424",
        height: 30,
        justifyContent: "center",
        alignItems: "center",
        flexDirection: "row",
    },
    offlineText: {
        color: "#fff",
    },
});

export default OfflineNotice;
