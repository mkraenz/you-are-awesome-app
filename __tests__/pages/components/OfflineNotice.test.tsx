import React from "react";
import "react-native";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import OfflineNotice from "../../../src/pages/components/OfflineNotice";

it("renders correctly", () => {
    const tree = renderer.create(<OfflineNotice />).toJSON();
    expect(tree).toMatchSnapshot();
});
