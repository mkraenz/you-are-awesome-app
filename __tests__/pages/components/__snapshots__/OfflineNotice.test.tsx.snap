// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#b52424",
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    No Internet Connection
  </Text>
</View>
`;
